test/e2e/e2e_test.go

```diff
@@ -252,45 +293,260 @@ var _ = Describe("e2e external gateway connectivity", func() {
+		if err != nil {
+			framework.Failf("Expected container %s running on %s error %v", ovnContainer, ovnWorkerNode, err)
+		}
+		for _, flowmod := range strings.Split(kubectlOut, "\n") {
+			// filter out irrelevant lines from ofctl output
+			if strings.Contains(flowmod, "n_packets=") {
+				// verify no flowmod counters were hit in br-ext
+				if !strings.Contains(flowmod, "n_packets=0") {
+					framework.Failf("Expected packets=0 but found the flow %s", flowmod)
+				}
+			}
+		}
+	})
+})
+
+// Test e2e inter-node connectivity over br-int
+var _ = Describe("test e2e inter-node connectivity between worker nodes", func() {
+	var haMode bool
+	svcname := "inter-node-e2e"
+	ovnNs := "ovn-kubernetes"
+	ovnWorkerNode := "ovn-worker"
+	ovnWorkerNode2 := "ovn-worker2"
+	ovnHaWorkerNode2 := "ovn-control-plane2"
+	ovnHaWorkerNode3 := "ovn-control-plane3"
+	ovnContainer := "ovnkube-node"
+	ovnNsFlag := fmt.Sprintf("--namespace=%s", ovnNs)
+	labelFlag := fmt.Sprintf("name=%s", ovnContainer)
+	jsonFlag := "-o=jsonpath='{.items..metadata.name}'"
+	f := framework.NewDefaultFramework(svcname)
 
-		// Add a macvlan interface to an ovnkube-node container
-		framework.Logf("Creating a macvlan interface named %s on pod %s", macvlanIface, ovnPodName)
-		framework.RunKubectlOrDie("exec", ovnPodName, ovnNsFlag, ovnContainerFlag,
-			"--", "ip", "link", "add", macvlanIface, "link", "eth0", "type", "macvlan", "mode", "bridge")
+	// Determine what KIND environment is running by querying the running nodes
+	BeforeEach(func() {
+		fieldSelectorFlag := fmt.Sprintf("--field-selector=spec.nodeName=%s", ovnWorkerNode)
+		fieldSelectorHaFlag := fmt.Sprintf("--field-selector=spec.nodeName=%s", ovnHaWorkerNode2)
 
-		// Assign an IPv4 address to the new macvlan interface
-		framework.Logf("Assigning IP address %s to %s", ovnTargetCidr, macvlanIface)
-		framework.RunKubectlOrDie("exec", ovnPodName, ovnNsFlag, ovnContainerFlag,
-			"--", "ip", "address", "add", ovnTargetCidr, "dev", macvlanIface)
+		// Determine if the kind deployment is in HA mode or non-ha mode based on node naming
+		kubectlOut, err := framework.RunKubectl("get", "pods", ovnNsFlag, "-l", labelFlag, jsonFlag, fieldSelectorFlag)
+		if err != nil {
+			framework.Failf("Expected container %s running on %s error %v", ovnContainer, ovnWorkerNode, err)
+		}
+		haMode = false
+		if kubectlOut == "''" {
+			haMode = true
+			kubectlOut, err = framework.RunKubectl("get", "pods", ovnNsFlag, "-l", labelFlag, jsonFlag, fieldSelectorHaFlag)
+			if err != nil {
+				framework.Failf("Expected container %s running on %s error %v", ovnContainer, ovnHaWorkerNode2, err)
+			}
+		}
+		// Fail the test if no pod is matched within the specified node
+		if kubectlOut == "''" {
+			framework.Failf("Unable to locate container %s on any known nodes", ovnContainer)
+		}
+	})
+
+	It("Should validate connectivity within a namespace of pods on separate nodes and ensure br-ext is not traversed", func() {
```

Comment:
how is this test different from line 305? It says br-ext is not traversed but I don't see it checked.